core/src/main/java/org/jdbi/v3/core/Query.java

```diff
@@ -79,89 +64,8 @@
-    }
-
-    /**
-     * Provide basic JavaBean mapping capabilities. Will instantiate an instance of resultType
-     * for each row and set the JavaBean properties which match fields in the result set.
-     *
-     * @param resultType JavaBean class to map result set fields into the properties of, by name
-     * @param <T> the JavaBean class to map result rows to.
-     *
-     * @return a Query which provides the bean property mapping
-     */
-    public <T> Query<T> mapToBean(Class<T> resultType)
-    {
-        return this.map(new BeanMapper<>(resultType));
-    }
-
-    /**
-     * Makes use of registered mappers to map the result rows to the desired type.
-     *
-     * @param resultType the type to map the query result rows to
-     * @param <T> the type to map result rows to
-     *
-     * @return a new query instance which will map to the desired type
-     *
-     * @see RowMappers#register(RowMapper)
-     * @see RowMappers#register(RowMapperFactory)
-     * @see Configurable#registerRowMapper(RowMapper)
-     * @see Configurable#registerRowMapper(RowMapperFactory)
-     */
-    @SuppressWarnings("unchecked")
-    public <T> Query<T> mapTo(Class<T> resultType)
-    {
-        return (Query<T>) this.mapTo((Type) resultType);
-    }
-
-    /**
-     * Makes use of registered mappers to map the result rows to the desired type.
-     *
-     * @param resultType the type to map the query result rows to
-     * @param <T> the type to map result rows to
-     *
-     * @return a new query instance which will map rows to the desired type
-     *
-     * @see RowMappers#register(RowMapper)
-     * @see RowMappers#register(RowMapperFactory)
-     * @see Configurable#registerRowMapper(RowMapper)
-     * @see Configurable#registerRowMapper(RowMapperFactory)
-     */
-    @SuppressWarnings("unchecked")
-    public <T> Query<T> mapTo(GenericType<T> resultType)
-    {
-        return (Query<T>) this.mapTo(resultType.getType());
-    }
-
-    /**
-     * Makes use of registered mappers to map the result rows to the desired type.
-     *
-     * @param resultType the type to map the query result rows to
-     *
-     * @return a new query instance which will map to the desired type
-     *
-     * @see RowMappers#register(RowMapper)
-     * @see RowMappers#register(RowMapperFactory)
-     * @see Configurable#registerRowMapper(RowMapper)
-     * @see Configurable#registerRowMapper(RowMapperFactory)
-     */
-    public Query<?> mapTo(Type resultType) {
-        return this.map(rowMapperForType(resultType));
-    }
-
-    public <T> Query<T> map(RowMapper<T> mapper)
-    {
-        return new Query<>(getConfig(),
-                getParams(),
-                mapper,
-                getHandle(),
-                getStatementBuilder(),
-                getSql(),
-                getContext(),
-                getStatementCustomizers());
+    public <R> R withResultSet(ResultSetCallback<R> callback) {
```

Comment:
This method comes from `ResultSetIterable`. The intent with the name was to match existing methods around the API, e.g. `Jdbi.withHandle(HandleCallback)`, `ResultIterable.withStream(StreamCallback)`, etc.

Originally I also had a `useResultSet` method but nothing ended up using it, so I torched it.

I'm open to a different name, but I personally lean toward the `with` method for the consistency with other methods that take callbacks.